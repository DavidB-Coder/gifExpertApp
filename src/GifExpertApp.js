import React, {useState} from 'react'
import { AddCategory } from './components/AddCategory';
import { GifGrid } from './components/GifGrid';

const GifExpertApp = ({defaultCategories=[]}) => {

    const [categories, setCategories] = useState(defaultCategories);

    return (
        <>
            <h2>GifExpertApp</h2>
            <AddCategory setCategory={setCategories}/>
            <hr/>            
            <ul>
                {
                    categories.map((category) => {
                        return (
                            <GifGrid 
                                category={category}
                                key={category}
                            />
                        )
                    })
                }
            </ul>
        </>
    )
}

export default GifExpertApp
