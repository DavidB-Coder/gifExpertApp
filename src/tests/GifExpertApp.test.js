import {shallow} from 'enzyme';
import '@testing-library/jest-dom';
import GifExpertApp from '../GifExpertApp';

describe('pruebas al componente GifExpertApp', () => {

    test('debe mostrar correctamente el componente', () => {

        const wrapper = shallow(<GifExpertApp/>);
        expect(wrapper).toMatchSnapshot();
        
    });

    test('debe de mostrar una lista de categorias', () => {
        const categories = ['perfect loops','pink panter']

        const wrapper = shallow(<GifExpertApp defaultCategories={categories}/>);
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('GifGrid').length).toBe(categories.length)

    });
    
});
