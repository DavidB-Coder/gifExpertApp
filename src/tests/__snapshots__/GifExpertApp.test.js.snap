// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas al componente GifExpertApp debe de mostrar una lista de categorias 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategory={[Function]}
  />
  <hr />
  <ul>
    <GifGrid
      category="perfect loops"
      key="perfect loops"
    />
    <GifGrid
      category="pink panter"
      key="pink panter"
    />
  </ul>
</Fragment>
`;

exports[`pruebas al componente GifExpertApp debe mostrar correctamente el componente 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategory={[Function]}
  />
  <hr />
  <ul />
</Fragment>
`;
